### PR TITLE
QA-336 Switch ci.skip for all branches synced from github.

### PR DESCRIPTION
**Pull request by merlin-northern**
ChangeLog:title
Signed-off-by: Peter Grzybowski <peter@northern.tech>

### CODE REVIEW

File: tests/tests/golden-files/test_push_cfengine.yml [hunk 1]
```diff
@@ -6,5 +6,5 @@ output:
 - 'git.Run: /usr/bin/git remote add gitlab git@gitlab.com:Northern.tech/CFEngine/website'
 - 'git.Run: /usr/bin/git fetch github'
 - 'git.Run: /usr/bin/git checkout -b master github/master'
-- 'git.Run: /usr/bin/git push -f gitlab master'
+- 'git.Run: /usr/bin/git push -f -o ci.skip gitlab master'
```
Comment (lluiscampos) [inline tests/tests/golden-files/test_push_cfengine.yml:9]: This means that no continuous deployment will happen :-1:

**Review by lluiscampos — CHANGES_REQUESTED** [review]

File: gitlab_sync.go [hunk 2]
```diff
@@ -51,9 +51,7 @@ func syncRemoteRef(log *logrus.Entry, org, repo, ref string, conf *config) error
 		}
 		// For the push, add option ci.skip for mender-qa
 		cmdArgs := []string{"push", "-f"}
-		if repo == "mender-qa" {
-			cmdArgs = append(cmdArgs, "-o", "ci.skip")
-		}
+		cmdArgs = append(cmdArgs, "-o", "ci.skip")
```
Comment (merlin-northern) [inline gitlab_sync.go:54]: @lluiscampos so in here we should set `ci.skip` for all the branches (GetHead()) excluding those (taken from the QA-336):
```
master
staging
production
[0-9]+\.[0-9]+\.x
pr_[0-9]+
```
right?

File: gitlab_sync.go [hunk 3]
```diff
@@ -2,13 +2,39 @@ package main
...
+		"production",
+	}
+	for _, n := range startByName {
+		if branchName == n {
+			return true
+		}
+	}
+
+	startByRegEx := []*regexp.Regexp{
+		regexp.MustCompile("^[0-9]+\\.[0-9]+\\."),
```
Comment (lluiscampos) [inline gitlab_sync.go:26]: Missing an `x`?
```suggestion
		regexp.MustCompile("^[0-9]+\\.[0-9]+\\.x"),
```

**Review by lluiscampos — APPROVED** [review]
Looks good! See comment.

I guess a hacker can still do a branch like `1.2.x-my-funny-branch` and get the pipeline running... but that is fine.

File: gitlab_sync.go (hunk 3 above)
Comment (merlin-northern) [reply gitlab_sync.go]: right!

Comment (merlin-northern) [reply gitlab_sync.go]: fixed. thank you.

**Comment by merlin-northern** [comment]
last pushes were just unit test case removal (wrt to the '"x"')

**Pull request merged**